src/dbos-executor.ts

```diff
@@ -907,34 +927,22 @@ export class DBOSExecutor implements DBOSExecutorContext {
           wCtxt.span.setAttribute('cached', true);
           wCtxt.span.setStatus({ code: SpanStatusCode.OK });
         } else if (err instanceof DBOSWorkflowCancelledError) {
+          wCtxt.span.setStatus({ code: SpanStatusCode.ERROR, message: err.message });
           internalStatus.error = err.message;
-          internalStatus.status = StatusString.CANCELLED;
-
-          if (!this.isDebugging) {
-            await this.systemDatabase.cancelWorkflow(workflowID);
+          if (err.workflowID === workflowID) {
+            internalStatus.status = StatusString.CANCELLED;
+            if (!this.isDebugging) {
+              await this.systemDatabase.cancelWorkflow(workflowID);
```

Comment:
What's going on here? We get the error that the workflow is cancelled, so we cancel it again?

Reply:
In the original pre-timeout code, we updated the status in the DB to CANCELLED directly. In the WF timeout PR, I changed that to call `cancelWorkflow`. I also call `cancelWorkflow` in `callPromiseWithTimeout` if the WF times out.

 It's not clear what the exact semantics of `DBOSWorkflowCancelledError` are, but we had code here to mark this WF as cancelled in sysdb at this point when I started on WF timeout PR. It's not clear if this error means the workflow *should* be cancelled (DB needs to be updated) or if if means the workflow *has* been cancelled (DB has already been updated). So I made `cancelWorkflow` idempotent. We were already checking WF status in `cancelWorkflow`, so it was easy to check for CANCELLED in addition to SUCCESS or ERROR.

Reply:
Removed this redundant call